File: php/Memory.php
<?php

include "Hardware.php";

class Memory extends Hardware{
	// atribut private
    private $frequency;
    private $memorySize;
    private $supportsCuda;

	// constructor
	public function __construct($frequency = "", $memorySize = "", $supportsCuda = ""){
        $this->frequency = $frequency;
        $this->memorySize = $memorySize;
        $this->supportsCuda = $supportsCuda;
	}

	// setter/getter
    public function setfrequency($frequency){
		$this->frequency = $frequency;
	}
    public function getfrequency(){
		return $this->frequency;
	}

    public function setmemorySize($memorySize){
		$this->memorySize = $memorySize;
	}
    public function getmemorySize(){
		return $this->memorySize;
	}

    public function setsupportsCuda($supportsCuda){
		$this->supportsCuda = $supportsCuda;
	}
    public function getsupportsCuda(){
		return $this->supportsCuda;
	}
}

?>

File: python/Memory.py
from Hardware import Hardware

class Memory(Hardware):
	# atribut private
	frequency = ""
	memorySize = ""
	supportsCuda = ""

	# getter/setter
	def setfrequency(self, frequency):
		self.frequency = frequency

	def getfrequency(self):
		return self.frequency

	def setmemorySize(self, memorySize):
		self.memorySize = memorySize

	def getmemorySize(self):
		return self.memorySize
	
	def setsupportsCuda(self, supportsCuda):
		self.supportsCuda = supportsCuda

	def getsupportsCuda(self):
		return self.supportsCuda
